include/dca/parallel/util/get_workload.hpp

```diff
@@ -42,6 +42,34 @@ int getWorkload(const unsigned int total_work, const unsigned int n_local_worker
   return getWorkload(local_work, n_local_workers, local_id);
 }
 
+inline void getComputeRange(const int& my_rank, const int& mpi_size,
+                            const uint64_t& total_G4_size, uint64_t& start, uint64_t& end) {
+  uint64_t offset = 0;
+  // check if originally flattened one-dimensional G4 array can be equally (up to 0) distributed across ranks
+  // if balanced, each rank has same amount of elements to compute
+  // if not, ranks with (rank_id < nb_more_work_ranks) has to compute 1 more element than other ranks
+  bool balanced = (total_G4_size % static_cast<uint64_t>(mpi_size) == 0);
+  uint64_t local_work = total_G4_size / static_cast<uint64_t>(mpi_size);
+
+  if(balanced) {
+        offset = static_cast<uint64_t>(my_rank)  * local_work;
+        end  = offset + local_work;
+  }
+  else {
+    int nb_more_work_ranks = total_G4_size % static_cast<uint64_t>(mpi_size);
```

Comment:
our coding standard generally bans prefix for type or unexplained prefixes.  It might just be a low coffee condition but what does nb mean?

Reply:
Agreed. What I tried to say is total number of ranks that need more work than others. Or maybe rename it to `n_more_work_ranks` or shall I write some comments around this variable?

Reply:
I think more_work_ranks is fine.

Reply:
Ok, just fixed. Thanks